server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -586,30 +586,47 @@ void sc_osc_handler::open_udp_socket(udp const& protocol, unsigned int port) {
 }
 
 bool sc_osc_handler::open_socket(int family, int type, int protocol, unsigned int port) {
-    if (protocol == IPPROTO_TCP) {
-        if (type != SOCK_STREAM)
-            return false;
-
-        if (family == AF_INET)
-            open_tcp_acceptor(tcp::v4(), port);
-        else if (family == AF_INET6)
-            open_tcp_acceptor(tcp::v6(), port);
-        else
-            return false;
-        return true;
-    } else if (protocol == IPPROTO_UDP) {
-        if (type != SOCK_DGRAM)
-            return false;
+    try {
```

Comment:
wasn't sure where to place this block -- maybe a shared function called by `open_tcp_acceptor` and `open_udp_socket` would be better?